tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/LraResource.java

```diff
@@ -410,6 +415,34 @@ public Response timeLimit(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId) {
         return Response.status(Response.Status.OK).entity(lraId.toASCIIString()).build();
     }
 
+    /**
+     * @see org.eclipse.microprofile.lra.tck.TckTests#timeLimitWithPreConditionFailed
+     */
+    @GET
+    @Path(TIME_LIMIT_HALF_SEC)
+    @Produces(MediaType.APPLICATION_JSON)
+    @LRA(value = LRA.Type.REQUIRED, timeLimit = 500, timeUnit = ChronoUnit.MILLIS)
+    public Response timeLimitTest2(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId) {
+        assertHeaderPresent(lraId, LRA_HTTP_CONTEXT_HEADER);
+
+        activityStore.add(new Activity(lraId));
+
+        try {
+            Thread.sleep(1000); // sleep for longer than specified in the timeLimit annotation attribute
+            // force the implementation to notice that the LRA should have timed out
+            lraTestService.waitForCallbacks(lraId);
+            // the next request should fail with a 412 code since the LRA should no longer be active
+            restPutInvocation(lraId, MANDATORY_LRA_RESOURCE_PATH, "");
+        } catch (WebApplicationException wae) {
+            return wae.getResponse();
```

Comment:
```suggestion
            return Response.status(wae.getResponse().getStatus()).build();
```

I tested this locally and the response returned in `wae` from `restPutInvocation` is actually closed in this method [1] which fails the underlying JAX-RS implementation at least in Narayana.

[1] https://github.com/eclipse/microprofile-lra/blob/master/tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/LraResource.java#L327

Reply:
Thanks. I pushed your recommended fix.